hazelcast-sql/src/test/java/com/hazelcast/jet/sql_nightly/SqlSTSInnerEquiJoinFaultToleranceStressTest.java

```diff
@@ -0,0 +1,282 @@
+
+            assertThat(duplicates)
+                    .as("Non-unique result count: " + duplicates.size())
+                    .isEmpty();
+        }
+
+        for (int i = firstItemId; i <= lastItemId; ++i) {
+            String key = "value-" + i;
+            assertThat(resultSet.remove(key))
+                    .as("Missing element: " + key)
+                    .isNotNull();
+        }
+
+        assertThat(resultSet)
+                .as("Unexpected items in the result: " + resultSet)
+                .isEmpty();
+    }
+
+    protected String setupFetchingQuery() {
+        return "CREATE JOB " + JOB_NAME +
+                " OPTIONS ('processingGuarantee'='" + processingGuarantee + "', 'snapshotIntervalMillis' = '1000')" +
+                " AS SINK INTO " + sinkTopic +
+                " SELECT s1.__key, s2.this FROM s1 JOIN s2 ON s1.__key = s2.__key";
+    }
+
+    class JobRestarter extends Thread {
+        private final JetService jetService;
+        private final JetServiceBackend jetBackend;
+        private volatile boolean finished;
+
+        JobRestarter(HazelcastInstance hazelcastInstance) {
+            this.jetService = hazelcastInstance.getJet();
+            this.jetBackend = getJetServiceBackend(hazelcastInstance);
+        }
+
+        @Override
+        public void run() {
+            try {
+                // wait for job start
+                assertTrueEventually(() -> assertNotNull(jetService.getJob(JOB_NAME)));
+
+                @SuppressWarnings("rawtypes")
+                AbstractJobProxy job = (AbstractJobProxy) jetService.getJob(JOB_NAME);
+                assertNotNull(job);
+
+                Long lastExecutionId = null;
+                while (!finished) {
+                    waitForNextSnapshot(jetBackend.getJobRepository(), job.getId(), SNAPSHOT_TIMEOUT_SECONDS, true);
+                    job.restart(restartGraceful);
+                    lastExecutionId = assertJobRunningEventually(instance(), job, lastExecutionId);
+                }
+            } catch (NullPointerException e) {
+                System.err.println(e);
+            } catch (Throwable e) {
+                logger.warning(null, e);
+                ex = e;
+            }
+        }
+
+        public void finish() {
+            this.finished = true;
+        }
+    }
+
+    private void createTopicData(SqlService sqlService, String topicName) {
+        try {
+            int itemsSank = 0;
+            for (int sink = 1; sink <= SINK_COUNT; sink++) {
+                StringBuilder queryBuilder = new StringBuilder("INSERT INTO " + topicName + " VALUES ");
+                for (int i = 0; i < EVENTS_PER_SINK; ++i) {
+                    ++itemsSank;
+                    queryBuilder.append("(").append(itemsSank).append(", 'value-").append(itemsSank).append("'),");
+                }
+                queryBuilder.setLength(queryBuilder.length() - 1);
+
+                assertEquals(itemsSank, EVENTS_PER_SINK * sink);
```

Comment:
this assert does not check anything (for loop?)

Reply:
Sanity check during development :)